src/configurator.rb

```diff
@@ -131,33 +112,24 @@ def group_cache
     # saved. If there is an old_answer then it is the default. In this case
     # it needs to be saved again so it is not lost.
     def ask_questions
-      questions.with_object({}) do |(node_q, content), memo|
-        next unless ask_question_based_on_parent_answer(node_q)
-        raw_answer = content.question.ask(highline)
-        content.answer = if raw_answer == content.default
-                           content.old_answer.nil? ? nil : answer
-                         else
-                           raw_answer
-                         end
-        memo[content.identifier] = content.answer unless content.answer.nil?
-      end
-    end
+      memo = {}
+      section_question_tree.ask_questions do |node_q, idx|
+        identifier = node_q.identifier
+        question = node_q.create_question
 
-    def ask_question_based_on_parent_answer(node_q)
-      # TODO: Use the filter methods on QuestionTree to remove the following
-      # logic. The old Tree::TreeNode structure has non-question structural
-      # nodes within it that needed to be manually filtered
-      # However the new QuestionTree object has filtered enumerators that
-      # remove the need for checking if a node is a question
-      if node_q.question? && !node_q.parent.question?
-        true
-      # Conditionally ask the question if the parent answer is truthy
-      elsif node_q.parent.content.answer
-        true
-      # Otherwise don't ask the question
-      else
-        false
+        question.default = default_hash[identifier]
+        question.progress_indicator = progress_indicator(idx)
+        question.old_answer = old_answers[identifier]
```

Comment:
Most of those properties have been removed in a future commit. `old_answer` has been removed completely as its only purpose was to form a tautology with the namespace answers. The progress indicator is determined by `QuestionTree` now so it is an input again.

The only one that is still an accessor is the `default` which is used by highline. This is because `create_question` has been hidden from `Configurator` to prevent coupling between `Configurator` and `Question`. The only exception to this is the `default`, hence why it still has its accessor.

Reply:
Fair enough, things are much nicer now with all the refactoring. I still think it may be better if the `default` was provided when creating the `Question`, but it's not such a big deal as the `default` isn't required to be set for a valid `Question` to be created so there isn't the same possibility of a semi-initialized question being created that I mentioned. We can always change it from an accessor later.